### PR TITLE
doc(modules): add modules guide

### PR DESCRIPTION
**Pull request by oojimmy**

### CODE REVIEW

File: modules/README.md [hunk 1]
```diff
@@ -2,3 +2,260 @@
 
 Modules stores the pre-built modules.
 Each module has BCC C file, and wasm C file, and build files for building WASM.
+
+## Writing modules
+The main purpose of modules is to provide a way to write eBPF and parser code, and then
+compile it to WASM, and then deploy it to the agent.
+
+Startship modules are composed of three parts: eBPF code, WASM code, and module definition.
+* eBPF: Ebpf side code can be written by all major frontend languages of writing eBPF programs,  including:
```
Comment (nascentcore-eng) [inline modules/README.md:11]: eBPF not Ebpf
PS: I'll see how to add a spell check into linter

Comment (nascentcore-eng) [reply modules/README.md:11]: https://github.com/tricorder-observability/starship/issues/91

File: modules/README.md [hunk 2]
```diff
@@ -2,3 +2,260 @@
...
+    * [BCC](https://github.com/iovisor/bcc)
+    * [BPFtrace](https://github.com/iovisor/bpftrace)
+    * Rust ([readbpf](https://github.com/foniod/redbpf)[aya](https://github.com/aya-rs/aya))
```
Comment (nascentcore-eng) [inline modules/README.md:14]: Remove
We don't support them yet

File: modules/README.md [hunk 3]
```diff
@@ -2,3 +2,260 @@
...
+* WASM: WASM side code can be written by any language that can compile to WASM, including:
+    * C
+    * Rust
+    * Go
+    * AssemblyScript
+    * etc.
```
Comment (nascentcore-eng) [inline modules/README.md:20]: WASM only support C/C++ right now

State truthfully

File: modules/README.md [hunk 4]
```diff
@@ -2,3 +2,260 @@
...
+  float F;
+  char C;
+  double D;
+  int I;
+  long long int L;
+  short Comm[TASK_COMM_LEN];
+};
+```
+
+
```
Comment (nascentcore-eng) [inline modules/README.md:103]: One blank lines

File: modules/README.md [hunk 5]
```diff
@@ -2,3 +2,260 @@
...
+### event_bindgen.h
+event struct definition for WASM. It was generated by [bindgen](https://github.com/eunomia-bpf/c-struct-bindgen) from event.h by running the following command:
```
Comment (nascentcore-eng) [inline modules/README.md:105]: Refer to the forked repo not the original one

File: modules/README.md [hunk 6]
```diff
@@ -2,3 +2,260 @@
...
+```shell
+ecc event.h --header-only && struct-bindgen event.bpf.o -p >event_bindgen.h
```
Comment (nascentcore-eng) [inline modules/README.md:107]: Link to the readme for generating event binding struct header, I think it's somewhere in one of the readme.md file, which has instructions for installing ECC and other tool

File: modules/README.md [hunk 7]
```diff
@@ -2,3 +2,260 @@
...
+    * etc.
+* module definition: including module name, module description, module version, module deployment information, etc.
+
+
+
+## Example
+As an example, the `sample_event` module has the following structure:
+
+```shell
+➜  sample_event git:(doc/modules) ✗ ls -al
```
Comment (JaredTan95) [inline modules/README.md:29]: how about list by `tree` view?

**Comment by nascentcore-eng** [comment]
We decide to move this doc into Feishu doc and for internal editing before publishing for review.

**Pull request closed without merging**